griptape/tools/base_tool.py

```diff
@@ -95,18 +96,23 @@ def schema(self) -> dict:
         return full_schema.json_schema(f"{self.name} ToolAction Schema")
 
     def activity_schemas(self) -> list[Schema]:
-        return [
-            Schema(
-                {
-                    Literal("name"): self.name,
-                    Literal("path", description=self.activity_description(activity)): self.activity_name(activity),
-                    **self.activity_to_input(
-                        activity,
-                    ),  # Unpack the dictionary in order to only add the key-values if there are any
-                },
-            )
-            for activity in self.activities()
-        ]
+        schemas = []
+
+        for activity in self.activities():
+            schema_dict: dict[Literal | schema.Optional, Any] = {
+                Literal("name"): self.name,
+                Literal("path", description=self.activity_description(activity)): self.activity_name(activity),
+            }
+
+            activity_schema = self.activity_schema(activity)
+            if activity_schema is not None:
+                schema_dict[Literal("input")] = activity_schema.schema
+            else:
+                schema_dict[schema.Optional("input")] = {}
+
+            schemas.append(Schema(schema_dict))
+
+        return schemas
```

Comment:
Faced some really weird type errors when doing the dictionary unpacking. Refactoring to imperative fixed.

Reply:
See https://github.com/griptape-ai/griptape/pull/1015/files#r1695354991, out of scope for this PR but definitely something we should do.